### ISSUE TITLE
ErrorException : rename(): No such file or directory while building a PHAR archive

### ISSUE THREAD

**Issue by svallamsetti** [laravel-zero/laravel-zero#159]
When I ran this command **php weather app:build** 
I got this exception  ErrorException  : rename(/vagrant/code/testApp/weather.phar,/vagrant/code/testApp/builds/weather): No such file or directory.

I believe when we named our application there is a file being created with the application name in the root directory but not with .phar extension. Seems like the Filesystem::move() method which is calling rename() method is trying to access the .phar file and unable to find that file.

**Comment by nunomaduro** [comment]
Hey @svallamsetti 👋

Please run `php weather app:build -vvv` and show me the output.

**Comment by nunomaduro** [comment]
@svallamsetti Do you have news on this?

**Comment by svallamsetti** [comment]
@nunomaduro I got 

1. Moving application to production mode: ✔
sh: ./box: Permission denied
   2. Compile into a single file: ✔
and then it threw the same exception
ErrorException : rename(/vagrant/code/testApp/weather.phar,/vagrant/code/testApp/builds/weather): No such file or directory.

**Comment by nunomaduro** [comment]
@svallamsetti Please perform a `composer update laravel-zero/framework` and re-run the build again.

**Comment by svallamsetti** [comment]
@nunomaduro still same error. But it works on my other vagrant box.

**Comment by nunomaduro** [comment]
@svallamsetti No idea. You still have the issue?

**Comment by nikoheikkila** [comment]
@nunomaduro I'm having the same issue too with both PHP 7.1 and 7.2. Using Laravel Zero version 5.7.10. Mounting the Zero application to Docker container and running `php zeroapp app:build` outputs following error:

```sh
[2018-11-06 13:45:22] development.ERROR: rename(/app/zeroapp.phar,/app/builds/zeroapp): No such file or directory {"exception":"[obje
ct] (ErrorException(code: 0): rename(/app/zeroapp.phar,/app/builds/zeroapp): No such file or directory at /app/vendor/illuminate/file
system/Filesystem.php:203)
```

Working directory of application is `/app` for your information. Permissions should be fine since the command is ran by root user inside container.

```sh
bash-4.4# ls -lha builds/
total 0
drwxr-xr-x    2 root     root          64 Oct 24 09:14 .
drwxr-xr-x   40 root     root        1.3K Nov  6 13:48 ..
```

**Comment by nikoheikkila** [comment]
I seem to have resolved this by copying the `box.json` file to my project directory. I used this: https://github.com/laravel-zero/laravel-zero/blob/stable/box.json

My guess is the app was reading some old configuration from somewhere – maybe a cache? Can you @svallamsetti try using this file if it does the trick for you too?

**Comment by svallamsetti** [comment]
@nikoheikkila That didn't help.

**Comment by svallamsetti** [comment]
> @svallamsetti No idea. You still have the issue?
yes

**Comment by WangNingkai** [comment]
I also have the same problem，beacause it can not create the `.phar` file.

**Comment by dabernathy89** [comment]
FWIW, I was running into this error, and it turned out to be invalid JSON in my `box.json` file (discovered by running the build with the `-vvv` flag).

**Comment by zajinx** [comment]
i found another fix for this issue on windows, using the latest version of laravel-zero and box

1. install box as global:  composer global require humbug/box
2. edit line 96 in vendor\laravel-zero\framework\src\Commands\BuildCommand.php
from 
```
'./box compile'.' --working-dir='.base_path().' --config='.base_path('box.json'),
```
to
```
'box compile'.' --working-dir='.base_path().' --config='.base_path('box.json'),
```

**Comment by nunomaduro** [comment]
Can you propose a pull request please ? On the develop branch.

**Comment by gravataLonga** [comment]
offtopic: tuga aqui.

Yep, the same thing happen to me, i don't have time to debug this. But i think is failing build the ".phar" object, because on directory there not exist a ".phar" file and the command is try to move a .phar file to  builds folder. May be this can give some hint.

**Comment by gravataLonga** [comment]
When running manually the ./box compile, i got the following error:  

`
vendor/laravel-zero/framework/bin/box compile --working-dir="/<path>/tool" --config="/<path>/tool/box.json" --debug
[debug] Checking BOX_ALLOW_XDEBUG
[debug] The xdebug extension is not loaded
[debug] Process restarting (BOX_ALLOW_XDEBUG=internal||1|*|*)
[debug] Running '/usr/local/Cellar/php/7.3.1/bin/php' '-n' '-c' '/private/var/folders/wv/yyrbf5rs1rn7hdhhv0vyfst00000gn/T/ZPW7vG' 'vendor/laravel-zero/framework/bin/box' 'compile' '--working-dir=/Users/jfontes/Sites/tool' '--config=/Users/jfontes/Sites/tool/box.json' '--debug' '--ansi'
[debug] Configured `phar.readonly=0`
[debug] Bumped the memory limit from "128M" to "512M"
[debug] Checking BOX_ALLOW_XDEBUG
[debug] Restarted (114 ms). The xdebug extension is not loaded

    ____
   / __ )____  _  __
  / __  / __ \| |/_/
 / /_/ / /_/ />  <
/_____/\____/_/|_|


Box version 3.3.1@d2a2a7f 2018-11-13 09:37:29 UTC


 // Loading the configuration file "/Users/jfontes/Sites/tool/box.json".


 [ERROR] The configuration file is invalid.
`

**Comment by gravataLonga** [comment]
After ones minutes to debug, the solutions was the command:  

```  
php <app> app:rename
```  

Not change the composer.json, the solutions is more ease than look! 👍

**Comment by nunomaduro** [comment]
This issue was been fixed on Laravel Zero 5.8.
